src/script_opt/ZAM/Ops.in

```diff
@@ -770,10 +770,9 @@ eval	auto op1 = frame[z.v1].ToVal(z.t);
 # the main instruction type, as always.
 
 macro EvalVal2InTableCore(op1, op2)
-	ListVal lv(TYPE_ANY);
-	lv.Append(op1);
-	lv.Append(op2);
-	ListValPtr lvp = {NewRef{}, &lv};
```

Comment:
😮

Reply:
Whoops, that was something for Vern. I'll drop it before I merge.

Reply:
Yeah, that one is fun. You get a bunch of repeated warnings about `lv` not being initialized when `lvp` is destructed because `lv` went out of scope long before that. You would think that `IntrusivePtr` should handle that correctly, but the compiler doesn't think so. Anyways, it doesn't make a whole lot of sense to construct the object and then create an `IntrusivePtr` around it when you could just make the ptr directly.

Reply:
Isn't this code actually broken? Creating an IntrusivePtr from a stack allocated object/ptr has no happy end unless you call `.release()` somewhere? Not sure if this is happening.

Reply:
That was my reaction as well when I started looking at it. The related warning looks like this:

```
In file included from ../../src/include/zeek/Expr.h:17,
                 from ../../src/include/zeek/script_opt/ScriptOpt.h:11,
                 from ../../src/script_opt/ZAM/ZBody.cc:11:
In destructor 'virtual zeek::ListVal::~ListVal()',
    inlined from 'void zeek::Unref(Obj*)' at ../../src/include/zeek/Obj.h:169:16,
    inlined from 'zeek::IntrusivePtr<T>::~IntrusivePtr() [with T = zeek::ListVal]' at ../../src/include/zeek/IntrusivePtr.h:117:22,
    inlined from 'zeek::ValPtr zeek::detail::ZBody::DoExec(zeek::detail::Frame*, zeek::detail::StmtFlowType&)' at ./ZAM-EvalDefs.h:5820:2:
../../src/include/zeek/Val.h:643:5: error: 'void operator delete(void*, std::size_t)' called on unallocated object 'lv' [-Werror=free-nonheap-object]
  643 |     ~ListVal() override = default;
      |     ^
In file included from ../../src/script_opt/ZAM/ZBody.cc:298:
./ZAM-EvalDefs.h: In member function 'zeek::ValPtr zeek::detail::ZBody::DoExec(zeek::detail::Frame*, zeek::detail::StmtFlowType&)':
./ZAM-EvalMacros.h:47:17: note: declared here
   47 |         ListVal lv(TYPE_ANY); \
      |                 ^~
./ZAM-EvalMacros.h:57:9: note: in expansion of macro 'EvalVal2InTableCore'
   57 |         EvalVal2InTableCore(frame[z.op1].ToVal(z.t2), frame[z.op2].ToVal(tt_ind[1]))
      |         ^~~~~~~~~~~~~~~~~~~
./ZAM-EvalDefs.h:5818:9: note: in expansion of macro 'EvalVal2InTablePre'
 5818 |         EvalVal2InTablePre(v1,v2,v3)
      |         ^~~~~~~~~~~~~~~~~~
```

Reply:
Yeah, I think it's been working by luck because the IntrusivePtr was deconstructed before the stack one, so nothing actually wound up getting free'd.

Reply:
Oh! Because the stack allocated object starts with ref count 1 and the NewRef made it 2, so it was never freed by ref-counting means. So not broken, but non-obvious and possibly dangerous.